### PR TITLE
chore: fixing running task in a proper zone preserving the active zone

### PR DESCRIPTION
**Pull request by obecny**
## Which problem is this PR solving?

- Partly fixes this:
https://github.com/open-telemetry/opentelemetry-js/issues/1206

For full fix another PR is needed from sdk
https://github.com/open-telemetry/opentelemetry-js/pull/1209

## Short description of the changes

- it run the task in active zone preserving the spawned zone instead of root zone

### CODE REVIEW

**Comment by obecny** [comment]
@open-telemetry/javascript-maintainers  ^^

File: plugins/web/opentelemetry-plugin-user-interaction/src/userInteraction.ts [hunk 1]
```diff
@@ -313,28 +313,31 @@ export class UserInteractionPlugin extends BasePlugin<unknown> {
...
-                const currentZone = Zone.current;
-                task._zone = currentZone;
-                return original.call(currentZone, task, applyThis, applyArgs);
-              });
-            } finally {
-              plugin._decrementTask(span);
-            }
+            return activeZone.run(() => {
+              try {
+                return plugin._tracer.withSpan(span as api.Span, () => {
```
Comment (mayurkale22) [inline plugins/web/opentelemetry-plugin-user-interaction/src/userInteraction.ts:323]: Why do you have to do Type Assertion here (`span as api.Span`)?

File: plugins/web/opentelemetry-plugin-user-interaction/test/userInteraction.test.ts [hunk 2]
```diff
@@ -40,6 +40,21 @@ import {
 const FILE_URL =
   'https://raw.githubusercontent.com/open-telemetry/opentelemetry-js/master/package.json';
 
+// this is needed until this is merged and released
+// https://github.com/open-telemetry/opentelemetry-js/pull/1209
```
Comment (mayurkale22) [inline plugins/web/opentelemetry-plugin-user-interaction/test/userInteraction.test.ts:44]: Do you think we need to open issue to track this later?

File: plugins/web/opentelemetry-plugin-user-interaction/package.json [hunk 3]
```diff
@@ -46,6 +46,7 @@
     "@opentelemetry/context-zone-peer-dep": "0.8.3",
     "@opentelemetry/plugin-xml-http-request": "0.8.3",
     "@opentelemetry/tracing": "0.8.3",
+    "@opentelemetry/context-base": "0.8.3",
```
Comment (mayurkale22) [inline plugins/web/opentelemetry-plugin-user-interaction/package.json:49]:
```suggestion
    "@opentelemetry/context-base": "0.9.0",
```

File: plugins/web/opentelemetry-plugin-user-interaction/test/userInteraction.test.ts (hunk 2 above)
Comment (dyladan) [reply plugins/web/opentelemetry-plugin-user-interaction/test/userInteraction.test.ts:44]: 1209 is merged and released now actually

Comment (dyladan) [reply plugins/web/opentelemetry-plugin-user-interaction/test/userInteraction.test.ts:44]: https://github.com/open-telemetry/opentelemetry-js/releases/tag/v0.9.0

File: plugins/web/opentelemetry-plugin-user-interaction/src/userInteraction.ts (hunk 1 above)
Comment (dyladan) [reply plugins/web/opentelemetry-plugin-user-interaction/src/userInteraction.ts:323]: should not be required as the assertion on line 319 should remove `undefined` from the type

Comment (obecny) [reply plugins/web/opentelemetry-plugin-user-interaction/src/userInteraction.ts:323]: this is a compiler error / issue. the internal _createSpan returns api.Span or undefined. Even though there is `if` check for span existence the compiler still complains. If you have a better idea how to change it I'm open for suggestion.

**Comment by owais** [comment]
Thank you for getting this fixed @obecny. Can we release a new version for this package any time soon?

**Comment by dyladan** [comment]
Yes I'll try to cut this today or tomorrow

**Pull request merged**